### PR TITLE
Esm dependency analysis script

### PR DESCRIPTION
**Pull request by LilyL0u**
There may be some issues with our version dates being accurate so I am looking into that

Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-618
As part of https://jira.dev.bbc.co.uk/browse/WSTEAMA-617

ESM dependencies - Upgrade now, delay or replace?

- Find out which dependencies we have need ESM upgrades. THIS PR DOES THIS.
- Is ESM blocking any of these upgrades?
- Prioritise these. How important is it to upgrade them. If a lot are ESM dependent and important, we might need to do it.  e.g Emotion
- Given this information, what do we want to do? Delay to wait for the ecosystem to catch up? But at risk of a dependency having a security patch. Get rid of those dependencies (e.g chalk was easily replaced)? Replace the ESM dependencies, until ESM is easier and better? Commit to doing ESM upgrades, because this is the future? At least start trying? If we decide that, how will we do this? Decision will be affected by the dependency analysis.
- Architectual decision records in simorgh-infrastructure. When a decision is made about this, we need to record it there. This can be revisited, based on how the ecosystem goes on.
- Upgrade, delay or replace decision for each dependency, or blanket decision over all.
- End artifact will be the architectural decision record, a dependency analysis (table?), and a plan.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Instructions for esmDependencyCheck.js
```
Comment (eagerterrier) [inline scripts/README.md]: can we have that as an h2? e.g. `## Instructions for... `

File: scripts/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+This script was written with the intent to have a way to quickly check if any of our dependencies are ESM (instead of CommonJS), and with that knowledge make the decision if and when to upgrade or replace them based on the fact changing to ESM can be difficult. It may become less difficult in the future as the ecosystem matures.
+
+In order to use this script, you need to put your GIT_TOKEN in your .env file. 
+
+This code 'const dotenv = require('dotenv').config();' brings that config into the script file so it can then extract the token with the lines 'const gitToken = dotenv.parsed.GIT_TOKEN || false;' and then use it to access information on github repositories. Without this included in your .env, the gitToken will be false and the script will not run. This is because without it the data will all come back undefined, and the script is useless. For this to work in an automated way in the pipeline, we will need to add environment variables with cosmos.
+
+For information on how to get your personal access git token to use for this, see 'https://docs.github.com/en/enterprise-server@3.6/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens'
+
+
```
Comment (eagerterrier) [inline scripts/README.md:13]: can we add a line here on how to run this script? e.g. `yarn run esmDependencyCheck`

File: scripts/esmDependencyCheck.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      }
+
+      let csvContents =
+        'Dependency,Type, Most Recent Version, Our Version, Our Version Freshness in Days';
+      dependencyTable.forEach(
+        ({ name, type, mostRecentVersion, ourVersion, ourFreshnessInDays }) => {
+          csvContents += `\n${name},${type},${mostRecentVersion},${ourVersion},${ourFreshnessInDays}`;
+        },
+      );
+      fs.writeFileSync('./esmDependencyTable.csv', csvContents);
```
Comment (eagerterrier) [inline scripts/esmDependencyCheck.js]: why doesn't the csv in this PR reflect the headers? Seems to be the dep freshness one (with downloads in it)

**Review by eagerterrier — COMMENTED** [review]
This is great, Lily

Left a few Qs

File: scripts/README.md (hunk 2 above)
Comment (LilyL0u) [reply scripts/README.md:13]: This is very important! Yes!

File: scripts/README.md (hunk 1 above)
Comment (LilyL0u) [reply scripts/README.md]: I have now learned that this is a thing that can be done. Thanks

File: scripts/esmDependencyCheck.js (hunk 3 above)
Comment (LilyL0u) [reply scripts/esmDependencyCheck.js]: eerrrr weird, will look into this. Wonder if I committed a version from earlier before I finished writing the script. I didn't push the new csv

Comment (LilyL0u) [reply scripts/esmDependencyCheck.js]: I have deleted it. When I include one (want to check the accuracy first) then a new one will be pushed.

**Pull request closed without merging**